pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFS.java

```diff
@@ -101,7 +100,7 @@ public boolean move(URI srcUri, URI dstUri, boolean overwrite)
       }
     } else {
       // ensures the parent path of dst exists.
-      URI parentUri = Paths.get(dstUri).getParent().toUri();
+      URI parentUri = dstUri.getPath().endsWith("/") ? dstUri.resolve("..") : dstUri.resolve(".");
```

Comment:
To me, this also has dependency on certain FS. Maybe better to add a `getParent()` API to `PinotFS`?